votingsystem/src/general_meeting.rs

```diff
@@ -0,0 +1,271 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pub use ctypes::{BlockId};
+pub use ckey::Ed25519Public as Public;
+use coordinator::context::SubStorageAccess;
+use coordinator::types::*;
+use parking_lot::RwLock;
+use coordinator::module::*;
+use crate::common::*;
+use foundry_module_rt::UserModule;
+use serde::{Deserialize, Serialize};
+use remote_trait_object::raw_exchange::{import_service_from_handle, HandleToExchange, Skeleton};
+use remote_trait_object::{service, Context as RtoContext, Service, ServiceRef};
+use rand::Rng;
+use std::sync::Arc;
+use rlp::Encodable;
+
+
+enum ExecuteError {
+    InvalidMetadata,
+    InvalidSign,
+    InvalidFormat,
+    InvalidAdmin,
+}
+
+struct Context{
+    pub storage: Option<Box<dyn SubStorageAccess>>,
+    pub admin: Option<Public>,
```

Comment:
How about reading `admin` in the BlockChain state always?
If the admin value in the state changes, the GeneralMeeting module must read the value.
Currently, we are not changing it, but I predict it will be changed.